tekton/pipeline_run_test.go

```diff
@@ -132,5 +164,55 @@ var _ = Describe("PipelineRun", func() {
 				To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
 		})
 
+		It("Can add ApplicationSnapshot as json to the PipelineRun", func() {
+			Expect(k8sClient.Create(ctx, snapshot)).Should(Succeed())
+
+			snapshot.TypeMeta.APIVersion = apiVersion
+			snapshot.TypeMeta.Kind = "ApplicationSnapshot"
+
+			releasePipelineRun.WithApplicationSnapshot(snapshot)
+
+			Expect(releasePipelineRun.Spec.Params[0].Name).To(Equal("applicationSnapshot"))
+			Expect(json.Unmarshal(
+				[]byte(releasePipelineRun.Spec.Params[0].Value.StringVal),
+				&unmarshaledApplicationSnapshotSpec)).Should(Succeed())
+
+			// check if the unmarshaled data has what we expect
+			Expect(unmarshaledApplicationSnapshotSpec.Application).To(Equal("testapplication"))
+		})
+
+		It("Can add the ReleaseStrategy to the PipelineRun", func() {
+			strategy.Spec.Pipeline = releasePipelineRun.AsPipelineRun().Name
+			releasePipelineRun.WithReleaseStrategy(strategy)
+			Expect(releasePipelineRun.Spec.PipelineRef.Name).
+				To(Equal(releasePipelineRun.AsPipelineRun().Name))
+		})
+
+		It("Can add a reference to the service account to be used", func() {
+			releasePipelineRun.WithServiceAccount(serviceAccountName)
+			Expect(releasePipelineRun.Spec.ServiceAccountName).To(Equal(serviceAccountName))
+		})
+
+		It("Can add a workspace to the PipelineRun using the given name and PVC", func() {
+			releasePipelineRun.WithWorkspace(workspace, persistentVolumeClaim)
+			Expect(releasePipelineRun.Spec.Workspaces[0].Name).To(Equal(workspace))
+			Expect(releasePipelineRun.Spec.Workspaces[0].PersistentVolumeClaim.ClaimName).
+				To(Equal(persistentVolumeClaim))
+		})
+
+		It("Is it a PipelineRun", func() {
+			Expect(isReleasePipelineRun(releasePipelineRun.
+				WithReleaseLabels(release.Name, release.Namespace).
+				AsPipelineRun())).To(Equal(true))
+		})
+
+		It("Can make sure if PipelineRun has has Succeeded", func() {
```

Comment:
Do we need this? This looks like a test I would expect to see in Tekton to be honest.

Reply:
The result of `hasPipelineSucceeded` is important for `setupControllerWithManager`. Check https://github.com/redhat-appstudio/release-service/blob/main/controllers/release/release_controller.go#L142

I assume it should behave properly.

Reply:
Understood, but in that case the this test should be in utils_test.go as it applies to any PipelineRun.

Reply:
Fair enough! I thought I had moved it already tbh! Moving it!

Reply:
wow, just noticed it is a dup! It is already in `utils_test.go`.